### PR TITLE
[v19] 08-Wasm Light Client

### PR DESCRIPTION
**Pull request by Reecepbcups**
Add the Wasm Client for Dot / Kusama connections. wasm is cool

## TODO
- [x]  my comments (use x/wasm or ibcwasm funcs)

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -404,7 +405,9 @@ func New(
 
 	if manager := app.SnapshotManager(); manager != nil {
 		err = manager.RegisterExtensions(
-			wasmkeeper.NewWasmSnapshotter(app.CommitMultiStore(), &app.AppKeepers.WasmKeeper),
+			// TODO: is this okay? (may not matter since we are sharing the VM instance)
+			// wasmkeeper.NewWasmSnapshotter(app.CommitMultiStore(), &app.AppKeepers.WasmKeeper),
+			wasmlckeeper.NewWasmSnapshotter(app.CommitMultiStore(), &app.AppKeepers.WasmClientKeeper),
```
Comment (Reecepbcups) [inline app/app.go:410]: Waiting on a resp from the IBC team about which to use. I would assume the x/wasm keeper

File: app/app.go [hunk 2]
```diff
@@ -438,7 +441,10 @@ func New(
 			tmos.Exit(err.Error())
 		}
 		ctx := app.BaseApp.NewUncachedContext(true, tmproto.Header{})
-		// Initialize pinned codes in wasmvm as they are not persisted there
+
+		// Initialize pinned codes in wasmvm as they are not persisted there.
+		// We do not use the wasm light client's impl since we do not want ALL to be pinned.
+		// The WasmKeeper will handle is as expected.
 		if err := app.AppKeepers.WasmKeeper.InitializePinnedCodes(ctx); err != nil {
```
Comment (Reecepbcups) [inline app/app.go:448]: Same here ^^

**Comment by Reecepbcups** [comment]
Support for this ends in March :'( oh well

**Review by joelsmith-2019 — APPROVED** [review]
Looks good

**Pull request merged**